app/src/main/java/com/bnyro/recorder/ui/components/PlayerController.kt

```diff
@@ -0,0 +1,185 @@
+                var tempSliderPosition by remember { mutableStateOf<Float?>(null) }
+                Slider(
+                    modifier = Modifier.weight(1f),
+                    value = tempSliderPosition ?: positionAndDuration.first.toFloat(),
+                    onValueChange = { tempSliderPosition = it },
+                    valueRange = 0f.rangeTo(
+                        positionAndDuration.second?.toFloat() ?: Float.MAX_VALUE
+                    ),
+                    onValueChangeFinished = {
+                        tempSliderPosition?.let {
+                            exoPlayer.seekTo(it.toLong())
+                        }
+                        tempSliderPosition = null
+                    }
+                )
+                Text(
+                    positionAndDuration.second?.let { DateUtils.formatElapsedTime(it / 1000) }
+                        ?: ""
+                )
+            }
+            ElevatedCard(
+                colors = CardDefaults.elevatedCardColors(
+                    containerColor = MaterialTheme.colorScheme.tertiaryContainer,
+                    contentColor = MaterialTheme.colorScheme.onTertiaryContainer
+                ),
+                shape = CircleShape
+            ) {
+                val playState by isPlayingState()
+                IconButton(
+                    onClick = {
+                        playPause()
+                    }
+                ) {
+                    when (playState) {
+                        PlayerState.Buffer -> {
+                            CircularProgressIndicator(modifier = Modifier.size(24.dp))
+                        }
+
+                        PlayerState.Play -> {
+                            Icon(
+                                Icons.Default.Pause,
+                                contentDescription = "Pause"
+                            )
+                        }
+
+                        PlayerState.Pause -> {
+                            Icon(
+                                Icons.Default.PlayArrow,
+                                contentDescription = "Play"
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun Player.isPlayingState(): State<PlayerState> {
+    return produceState(
+        initialValue = if (isPlaying) {
+            PlayerState.Play
+        } else {
+            PlayerState.Pause
+        },
+        this
+    ) {
+        val listener = object : Player.Listener {
+            override fun onIsPlayingChanged(isPlaying: Boolean) {
+                playbackState
+                value = if (isPlaying) {
+                    PlayerState.Play
+                } else {
+                    PlayerState.Pause
+                }
+            }
+        }
+        addListener(listener)
+        if (!isActive) {
+            removeListener(listener)
```

Comment:
That listener is a different object each time the functions is called. I think you intended to move the creation of `listener` outside of the function?

Reply:
Moving it outside the function is a bad idea. Outside function is a composable function and it may run at any time a re composition occurs. also by doing that we won't be able to change the `value` which is a part of `produceState` function.

creating the listener inside the `produceState` function is better because it is not a composable function

Reply:
![image](https://github.com/Bnyro/RecordYou/assets/64766434/498eb676-03a4-493c-be77-6d076a80a8a1)

Reply:
Right, mixed something up. Nevertheless the few lines after don't make much sense then.